File: akatpp/akat_sub.c
FUNCTION$(u8 ${sub_name}()) {
    COROUTINE$(need_return, state_type = ${state_type}) {
        ${body}
    }

    akat_coroutine_state = AKAT_COROUTINE_S_START;
akat_coroutine_l_end:
    return akat_coroutine_state;
}

File: akatpp/akat_sub.py
class Macro:
    def render(self, inv):
        ctx = akat.prepare(inv, required_args = ["sub_name"], optional_kvs = { "state_type": "u8"}, body = True, allow_nesting = True)
        return akat.transform(akat.render(self, body = inv.body, sub_name = ctx.sub_name, state_type = ctx.state_type))
